src/shell/xdg/frame/mod.rs

```diff
@@ -0,0 +1,145 @@
+//! The frame to use with XDG shell window.
+
+use std::num::NonZeroU32;
+
+use crate::reexports::client::protocol::wl_surface::WlSurface;
+use crate::reexports::protocols::xdg::shell::client::xdg_toplevel::ResizeEdge;
+
+use crate::shell::xdg::window::{WindowManagerCapabilities, WindowState};
+
+pub mod fallback_frame;
+
+/// The interface for the client side decorations.
+pub trait DecorationsFrame: Sized {
```

Comment:
Decorations feels redundant in the name imo

Reply:
I feel like you'd need to rename the trait most of the time, so I've added the prefix, if anything, it's in a `frame`, not in `decorations` module.